Add explanation on how to work with private repos (DVC+MLEM) (#312)
iterative/mlem.ai: content/docs/user-guide/dvc.md

Learn more about [DVC](https://dvc.org/doc) and how it can be useful for
training your ML models.

## Working with private repositories

> If you commit your models to a private repository and use DVC to store binaries,

Reviewer: ```suggestion
If you commit model metafiles to a private repo and use DVC to store binaries,
```

> you'll need to authorize both via SSH (for DVC) and via HTTPS (for MLEM).

Reviewer 2: authorize -> authenticate
Would be more accurate everywhere here imo

Author: Fixed.

SSH authorization is usually achieved by running `git push` against a SSH
remote, or it can be done with
[`gh auth login`](https://cli.github.com/manual/gh_auth_login) (if you use
Github).

HTTPS authorization is done by setting `GITHUB_USERNAME` and `GITHUB_TOKEN` env

> vars. You can generate a token [here](https://github.com/settings/tokens).

Reviewer 2: env vars -> environment variabls
Also it's not clear here if you need or not to generate a token ("can" ?)

Author: Yep, it's a limitation. I'll add few sentences.

Author: I don't really know why DVC doesn't work with HTTP authentication. Maybe @skshetry can briefly explain the reason.

Learn more about [DVC](https://dvc.org/doc) and how it can be useful for
training your ML models.

## Working with private repositories

If you commit model metafiles to a private repo and use DVC to store binaries,
you'll need to authenticate both via SSH and via HTTPS. SSH authentication is
required for DVC, since DVC shallow clones the repo underneath via SSH. MLEM
instead uses `fsspec`'s
[`GitHubFileSystem`](https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.implementations.github.GithubFileSystem)

> to access the repo, which uses HTTP and authenticate with it underneath.

Reviewer 2: ```suggestion
to access the repo, which uses HTTPS for authentication.
```

SSH authentication is usually achieved by running `git push` against a SSH

> remote, or it can be done with

Reviewer 2: ```suggestion
remote, or can be done using
```

[`gh auth login`](https://cli.github.com/manual/gh_auth_login) (if you use
Github).

HTTPS authentication is done by setting `GITHUB_USERNAME` and `GITHUB_TOKEN`
environment variables. You need to generate a token
[here](https://github.com/settings/tokens).

> It's is important to first authenticate with SSH and then with HTTPS. Otherwise,

Reviewer 2: ```suggestion
It's important to first authenticate with SSH, and only then with HTTPS. Otherwise,
```
